src/common/transformations/src/transformations/mlir/convert.cpp

```diff
@@ -269,6 +269,70 @@ class Partitioner : public ov::pass::ModelPass {
     }
 };
 
+struct ConvertRelu {
+    void operator()(ConversionContext& context, NodePtr node) {
+        auto loc = createLocation(context.context, node);
+        auto& builder = context.builder();
+        // TODO: Support broadcasts
+        const auto input = context.getInputs(node)[0];
+        const auto ov_output_element_type = node->get_output_element_type(0);
+        const auto ov_output_shape = node->get_output_partial_shape(0);
+        auto outType = importTensor(context.context, ov_output_shape, ov_output_element_type);
+        // Named unary ops directly overwrite data in `outs` buffer so, there is no need to provide non-empty
+        // destination at the tensor-level.
+        // Use `tensor.empty` to avoid temporary buffer allocation and memcpy after bufferization.
+        llvm::SmallVector<Value> dynamicSizes;
+        for (auto [idx, dim] : llvm::enumerate(outType.getShape())) {
+            if (!mlir::ShapedType::isDynamic(dim))
+                continue;
+            auto dimSize = builder.create<tensor::DimOp>(loc, input, idx);
+            dynamicSizes.push_back(dimSize);
+        }
+        auto empty = builder.create<tensor::EmptyOp>(loc, outType, dynamicSizes);
+        auto zero = getConstant(builder, ov_output_element_type, 0);
+        auto fill = builder.create<linalg::FillOp>(loc, mlir::ValueRange{zero}, mlir::ValueRange{empty});
+        auto relu =
+            builder.create<linalg::MaxOp>(loc, mlir::ValueRange{input, fill.getResult(0)}, mlir::ValueRange{empty});
+        context.addOutputs(node, relu);
+    }
+};
+
+bool elementwise_f32_unary_no_broadcast_predicate(const ov::Output<ov::Node>& output) {
+    if (output.get_element_type() != ov::element::f32) {
+        return false;
+    }
+    // Check if implicit broadcast is possible, reject in this case
+    // Relies on symbolic information -- register SymbolicPropagation before applying this pattern
+    auto input_shape = output.get_node_shared_ptr()->get_input_partial_shape(0);
+    auto output_shape = output.get_partial_shape();
+    if (output_shape.rank().is_dynamic() || input_shape.rank().is_dynamic()) {
+        return false;
+    }
+    if (output_shape.rank().get_length() != input_shape.rank().get_length()) {
+        return false;
+    }
+
+    for (size_t i = 0; i < output_shape.size(); ++i) {
+        if (output_shape[i] != input_shape[i]) {
+            return false;
+        }
+        // Continue if all shapes are static.
+        if (output_shape[i].is_static() && input_shape[i].is_static()) {
+            continue;
+        }
+        if (!ov::symbol::are_equal(output_shape[i].get_symbol(), input_shape[i].get_symbol())) {
+            return false;
+        }
+    }
+
+    return true;
+}
+
+template <typename Op>
+NodePtr elementwise_f32_unary_no_broadcast() {
+    using namespace ov::pass::pattern;
+    return wrap_type<Op>({any_input()}, elementwise_f32_unary_no_broadcast_predicate);
+}
```

Comment:
@adam-smnk, may I ask you to move the whole thing into a separate file under the `op` subdirectory as it was implemented for `MatMul`? If it is not very clear, you can refuse, no problem at all -- I'll do it by myself later and also will reorganizer the part that starts to be a boilerplate.

Reply:
No problem, I'll move it there.